src/sidebar/reducers/annotations.js

```diff
@@ -265,19 +264,16 @@ function addAnnotations(annotations, now) {
     var anchoringAnnots = added.filter(metadata.isWaitingToAnchor);
     if (anchoringAnnots.length) {
       setTimeout(function () {
-        arrayUtil
+        var statusUpdates = arrayUtil
           .filterMap(anchoringAnnots, function (annot) {
             return findByID(getState().annotations, annot.id);
           })
           .filter(metadata.isWaitingToAnchor)
-          .forEach(function (orphan) {
-            dispatch({
-              type: actions.UPDATE_ANCHOR_STATUS,
-              anchorTimeout: true,
-              id: orphan.id,
-              tag: orphan.$tag,
-            });
-          });
+          .reduce((statusUpdates, orphan) => {
+            statusUpdates[orphan.$tag] = 'timeout';
+            return statusUpdates;
+          }, {});
+        dispatch(updateAnchorStatus(statusUpdates));
```

Comment:
I have no idea what is happening here

Reply:
Thanks for the feedback. I rewrote this piece of code. Please let me know if it seems clearer now.